triple-store/triple-store-api/src/main/java/com/powsybl/triplestore/api/TripleStore.java

```diff
@@ -57,6 +57,8 @@
 
     Set<String> contextNames();
 
+    void copyFrom(TripleStore source);
```

Comment:
We did not want to write a `clone` method (and hence implement `Cloneable` interface), as it is discouraged (sonar rule: "clone" should not be overridden (squid:S2975)). 
Java interfaces can not enforce creation of copy constructors, so we require only that all triplestore  implementations provide some way of copying all contents from a given source (`copyFrom`) and then in the `TripleStoreFactoryService` we define a factory method for creating copies.
I would prefer to rename copyFrom to `add`: it would be uniform with the other `add` methods.